src/vmm/src/lib.rs

```diff
@@ -1093,4 +1128,98 @@ mod tests {
             assert!(fdt.write_to_mem(&vmm.guest_memory, fdt_offset).is_err());
         }
     }
+    #[test]
+    fn test_address_alloc() {
```

Comment:
To me this looks like a set of unit tests for the allocator itself, not the vmm integration testing. Therefore I do not think this belongs here; these kind of tests belong to the allocator crate itself.

Reply:
I would like us to keep parts of this tests, specifically the negative ones where we try to allocate something that does not fit in the MMIO range defined for each platform. That one would test that we initialized the AddressAllocator properly, but it's also a bit of a simplistic test that I was considering to be a "safety net" testing that the allocator behaves as we expect it to.